_drafts/fwup-signing.md

```diff
@@ -0,0 +1,639 @@
+---
+title: "Authenticate firmware builds with code signing"
+description: WIP
+author: francois
+---
+
+<!-- excerpt start -->
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+{% include toc.html %}
+
+## Firmware Signing Explained
+
+### What is code signing
+
+Code signing is a method of proving a file was created by a trusted source and
+has not been tampered with. This is achieved by creating a signature for the
+file: a token which can be verified but not forged.
+
+We all use signatures in our day to day life, by scribbling our name at the
+bottom of our credit card receipts, contracts, and bank checks. These signatures
+can be verified quite easily by comparing them to previous signatures, but are
+hard to forge because they rely on an individual's muscle memory.
+
+Code signing uses cryptographic algorithms to achieve a similar goal. The
+resulting signatures can only be generated by people who know a given password,
+also known as a secret key, but can be verified as genuine by anyone.
+
+Additionally, the signatures are tied to a given firmware binary so should the
+binary change by so much as 1 bit the signature would no longer be valid.
```

Comment:
I'm struggling with words today, but you jump right into firmware binaries without yet talking about firmware binaries. I think I'm missing how this all applies to firmware from a high level.

Something like:

When a signature is generated for a firmware binary, it is uniquely tied to that exact firmware. Changing the binary so much as 1 bit would produce a different signature.

With this in mind, if a device is given a signed binary and a signature and can verify both that the binary and signature are valid and that the binary was created by the correct author, then we know the binary has not been tampered with.


Basically, connect the dots for me, either in a diagram or words, before we get into this stuff where we are dealing directly with openssl

Reply:
Used your words, they're great